src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java

```diff
@@ -333,47 +363,60 @@ public void putHoldingsStorageHoldingsByHoldingsRecordId(
                 if (holdingsList.size() == 1) {
                   try {
                     postgresClient.startTx(connection -> {
-                      updateItemEffectiveCallNumbersByHoldings(connection, postgresClient, entity).setHandler(updateResult -> {
-                        if (updateResult.succeeded()) {
-                          postgresClient.update(connection, HOLDINGS_RECORD_TABLE, entity,
-                            "jsonb", String.format(WHERE_CLAUSE, holdingsRecordId), false,
-                            update -> {
-                              try {
-                                if (update.succeeded()) {
-                                  postgresClient.endTx(connection, done -> {
-                                    asyncResultHandler.handle(
-                                      Future.succeededFuture(
-                                        PutHoldingsStorageHoldingsByHoldingsRecordIdResponse
-                                          .respond204()));
-                                  });
-                                }
-                                else {
+                      final HoldingsRecord existingHoldings = holdingsList.get(0);
+                      if (Objects.equals(entity.getHrid(), existingHoldings.getHrid())) { 
```

Comment:
Does it make sense to move this validation right after `if (holdingsList.size() == 1) {` so no transaction will be started at all?